src/mlpack/core/data/split_data.hpp

```diff
@@ -61,20 +64,28 @@ void Split(const arma::Mat<T>& input,
   trainLabel.set_size(trainSize);
   testLabel.set_size(testSize);
 
-  const arma::Col<size_t> order =
-      arma::shuffle(arma::linspace<arma::Col<size_t>>(0, input.n_cols - 1,
-                                                      input.n_cols));
-
-  for (size_t i = 0; i != trainSize; ++i)
+  if (shuffleData)
   {
-    trainData.col(i) = input.col(order[i]);
-    trainLabel(i) = inputLabel(order[i]);
+    arma::Col<size_t> order;
+    order = arma::shuffle(arma::linspace<arma::Col<size_t>>(0,
+        input.n_cols - 1, input.n_cols));
```

Comment:
```suggestion
    arma::uvec order = arma::shuffle(arma::linspace<arma::uvec>(
        0, input.n_cols - 1, input.n_cols);
```

Can we use the old implementation?  (This happens a few times in the code.) It's *basically* the same but uses one fewer line, and the `const` is a good idea for the compiler.

I changed the type to `uvec` to match the next suggestion... :)

Reply:
Changed